flexget/plugins/clients/qbittorrent.py

```diff
@@ -135,9 +191,13 @@ def add_torrent_file(self, file_path, data, verify_cert):
             )
         logger.debug('Added torrent file {} to qBittorrent', file_path)
 
-    def add_torrent_url(self, url, data, verify_cert):
+    def add_torrent_url(self, url, data, verify_cert, hash_torrent):
         if not self.connected:
             raise plugin.PluginError('Not connected.')
+
+        if self.check_torrent_exists(url, verify_cert, hash_torrent):
```

Comment:
Maybe we could call this earlier, perhaps [hereabouts](https://github.com/Flexget/Flexget/blob/282181c34df1e5b0e41d642414d25d18303be066/flexget/plugins/clients/qbittorrent.py#L276) This would have the advantage of only needing to call it once whether file or magnet, and not having to pass info hash through the add_torrent_X functions.